### PR TITLE
fix: adding window check for server-side rendering

### PR DESCRIPTION
**Pull request by mathisscott**
• added check for server-side rendering
• fixed an issue with the dev-core app icon sets page

Signed-off-by: Scott Mathis <smathis@vmware.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [x] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #4240 

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

### CODE REVIEW

File: src/clr-core/common/utils/register.ts [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
+    }
   }
-});
+);
 
-export const registerElementSafely = addElementToRegistry(__, __, window.customElements);
+export function registerElementSafely(tagName: string, elementClass: any) {
+  const guard = and(!isNil(window), existsInWindow(['customElements']));
+
+  if (!guard) {
```
Comment (Shijir) [inline src/clr-core/common/utils/register.ts]: nit: can move that `addElementToRegistry` call below into this check but with `guard` equals truthy?

Comment (mathisscott) [reply src/clr-core/common/utils/register.ts]: sure. makes sense. thanks!

File: src/clr-core/common/utils/register.ts [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
+    } else {
+      registry.define(tagName, elementClass);
+    }
   }
-});
+);
 
-export const registerElementSafely = addElementToRegistry(__, __, window.customElements);
+export function registerElementSafely(tagName: string, elementClass: any) {
+  const guard = and(!isNil(window), existsInWindow(['customElements']));
```
Comment (coryrylan) [inline src/clr-core/common/utils/register.ts]: nit: Could this be extracted to a function like `isBrowser()`?

Is there an advantage of `and` vs `&&` if its not used as a curried func?

Comment (mathisscott) [reply src/clr-core/common/utils/register.ts]: Re: and. Nope. I can change that. 

Re: shared function. Yup. Will do.

Comment (mathisscott) [reply src/clr-core/common/utils/register.ts]: Re: and. Nope. I can change that. 

Re: shared function. Yup. Will do.

**Pull request merged**